transport.go

```diff
@@ -27,18 +27,26 @@ import (
 	"time"
 )
 
-func initTransport(upstreamCAFile string) (http.RoundTripper, error) {
+func initTransport(upstreamCAFile, upstreamClientCertPath, upstreamClientKeyPath string) (http.RoundTripper, error) {
 	if upstreamCAFile == "" {
 		return http.DefaultTransport, nil
 	}
 
-	rootPEM, err := os.ReadFile(upstreamCAFile)
+	upstreamCAPEM, err := os.ReadFile(upstreamCAFile)
```

Comment:
Is there a [apiserver / dynamiccertifcates](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates) solution we could leverage here?

I think for the aggregated certs, something like [this](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go#L264) is used.

Reply:
Yes. We should do that in a follow-up that would bring us closer to the apiserver serving structures.

Reply:
My other PR - https://github.com/brancz/kube-rbac-proxy/pull/210 - is there to make the transition to upstream option structures smoother

Reply:
I propose to address it along with `Reuse k8s.io/apiserver serving logic` which will likely bring these dynamic cert providers for other bundles, too.